ansible/deploy.yml

```diff
@@ -101,19 +116,13 @@
         when: epoch.stat.exists == True and ping.stdout == "pong"
         tags: [daemon]
 
-      - name: Register remote package
-        set_fact:
-          is_remote_package: true
-        when: "'http' in package"
-        tags: [package]
-
       - name: Extract epoch package into {{ project_root }}
         unarchive:
-          remote_src: "{{ is_remote_package | bool }}"
+          remote_src: true
           keep_newer: true
-          src: "{{ package }}"
+          src: "{{ packages_path }}/{{ package|basename }}"
           dest: "{{ project_root }}"
-        when: package is defined
+        when: remote_package.changed or local_package.changed
```

Comment:
Does this throw exception if using local package, on the basis that remote_package.changed is evaluated first and remote_package is unbound?

Reply:
I shall test it

Reply:
Works fine:
```
TASK [Download remote package] ***********************************************************************************************************************************************
Monday 10 September 2018  12:33:51 +0000 (0:00:00.538)       0:00:15.694 ****** 
skipping: [ae-dev1-epoch-n0]
skipping: [ae-dev1-epoch-n1]

TASK [Upload local package] **************************************************************************************************************************************************
Monday 10 September 2018  12:33:52 +0000 (0:00:00.594)       0:00:16.289 ****** 
changed: [ae-dev1-epoch-n1]
changed: [ae-dev1-epoch-n0]

TASK [Make sure /home/epoch/node exists] *************************************************************************************************************************************
Monday 10 September 2018  12:35:16 +0000 (0:01:23.949)       0:01:40.239 ****** 
ok: [ae-dev1-epoch-n0]
ok: [ae-dev1-epoch-n1]

RUNNING HANDLER [Check epoch binary] *****************************************************************************************************************************************
Monday 10 September 2018  12:35:17 +0000 (0:00:01.106)       0:01:41.345 ****** 
ok: [ae-dev1-epoch-n1]
ok: [ae-dev1-epoch-n0]

RUNNING HANDLER [Ping epoch] *************************************************************************************************************************************************
Monday 10 September 2018  12:35:18 +0000 (0:00:01.442)       0:01:42.787 ****** 
changed: [ae-dev1-epoch-n1]
changed: [ae-dev1-epoch-n0]

RUNNING HANDLER [Stop epoch] *************************************************************************************************************************************************
Monday 10 September 2018  12:35:22 +0000 (0:00:03.584)       0:01:46.372 ****** 
changed: [ae-dev1-epoch-n1]
changed: [ae-dev1-epoch-n0]

TASK [Extract epoch package into /home/epoch/node] ***************************************************************************************************************************
Monday 10 September 2018  12:35:32 +0000 (0:00:09.633)       0:01:56.005 ****** 
changed: [ae-dev1-epoch-n1]
changed: [ae-dev1-epoch-n0]

```